plugins/catalog/src/js/repositories/RepositoriesDelete.js

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+
+import { componentFromStream } from "data-service";
+import { Subject } from "rxjs/Subject";
+import { Observable } from "rxjs/Observable";
+import "rxjs/add/operator/combineLatest";
+import "rxjs/add/operator/startWith";
+import "rxjs/add/operator/map";
+import "rxjs/add/operator/catch";
+
+import { deleteRepository } from "./data/repositoriesStream";
+
+import RepositoriesDeleteConfirm from "./components/RepositoriesDeleteConfirm";
+import RepositoriesError from "./components/RepositoriesError";
+
+const getErrorMessage = (response = {}) => {
+  if (typeof response === "string") {
+    return response;
+  }
+
+  return response.description || response.message || "An error has occurred.";
+};
+
+const deleteEvent$ = new Subject();
+const deleteRepository$ = deleteEvent$
+  .switchMap(repository => {
+    return deleteRepository(repository.name, repository.url)
+      .startWith({ pendingRequest: true })
+      .map(result => {
+        return {
+          result,
+          pendingRequest: false
+        };
+      })
+      .do(() => {
+        repository.complete();
+      });
+  })
+  .startWith({ pendingRequest: false })
+  .catch(error => {
+    return Observable.of({
+      error: getErrorMessage(error.response),
+      pendingRequest: false
+    });
+  });
+
+const RepositoriesDelete = componentFromStream(props$ => {
```

Comment:
The problem here is: Imagine you have a stream of data (which is a good way to model that in a reactive application) and you want to render that date on the stream using react (which is a greatway to display things). How would you do it?

The straightforward approach would be to subscribe to the stream every time you mount the component, setting the component `state` to be the result of the stream, and having `render()`depending on the state. 

But if you want to separate state from rendering even further, you would like to use react `pure components` and handle the state outside react.

That is what we are trying to do that. To write a function that achieves the transformation of a stream of data, into a single component that is rendered with the most actual version of the data. `const myComponent = component(stream(DATA .... DATA ...... DATA))` 

We also have to add to that mix that there are also props coming from the parent (data down), and events from the children (events up). The first we implement with the props$ stream offered by recompose, the latter with solve with the creation of a `Subject()` that we also merge (on our data or our props ,does not really matter). Actually recompose have `createEventHandler` for that but we decided not to use it because it is too limiting on the Behaviour of the observable for events.

Reply:
Thanks, this was very helpful and will be helpful for others to read too. I will also watch that egghead course to hear more about this. 👍